task.go

```diff
@@ -143,24 +142,27 @@ func (t *task) Status(ctx context.Context) (TaskStatus, error) {
 // Wait is a blocking call that will wait for the task to exit and return the exit status
 func (t *task) Wait(ctx context.Context) (uint32, error) {
 	// TODO (ehazlett): add filtering for specific event
-	events, err := t.client.EventService().Stream(ctx, &eventsapi.StreamEventsRequest{})
+	eventstream, err := t.client.EventService().Stream(ctx, &eventsapi.StreamEventsRequest{})
 	if err != nil {
 		return UnknownExitStatus, err
 	}
 	<-t.pidSync
+evloop:
 	for {
-		evt, err := events.Recv()
+		evt, err := eventstream.Recv()
 		if err != nil {
 			return UnknownExitStatus, err
 		}
-		if evt.Event.TypeUrl == "types.containerd.io/containerd.v1.types.event.RuntimeEvent" {
-			e := &event.RuntimeEvent{}
-			if err := proto.Unmarshal(evt.Event.Value, e); err != nil {
+
+		switch {
```

Comment:
nit: couldn't we check `if events.Is(evt.Event, &eventsapi.RuntimeEvent{})` to remove the need for the `evloop` label?